### PR TITLE
Dse 996 dex

### PR DESCRIPTION
**Pull request by benshaw**
## Description
- Remove models and move them to the models package
- Add dex models as a dependency
- Run scalfmt and scalafix

## AC 
Move Models contained within Dex client to a common models package

## Source
https://dswift.atlassian.net/jira/software/projects/DSE/boards/5?selectedIssue=DSE-996

## Note
No functional changes are contained in this review.
Just moving of files, adding of dependencies and running of code linters

## Type
- [ ] Bug Fix
- [ ] Feature Addition 
- [X] Refactor
- [ ] HotFix

## Checklist
- [X] tested locally
- [X] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by dataswifty — APPROVED** [review]

**Comment by dataswifty** [comment]
> SonarCloud Quality Gate failed.

What's this?  Can we get it on all projects?

**Comment by benshaw** [comment]
> > SonarCloud Quality Gate failed.
> 
> What's this? Can we get it on all projects?

I think @pperzyna the magician added it

File: src/main/scala/org/hatdex/dex/api/services/DexOffers.scala [hunk 1]
```diff
@@ -11,22 +11,24 @@ package org.hatdex.dex.api.services
...
   val ws: WSClient
   val schema: String
   val dexAddress: String
 
-  import DexJsonFormats.offerClaimsInfoFormat
+  import io.dataswift.models.dex.json.DexJsonFormats._
 
+  @deprecated("This endpoint does not exist in dex",
+              since = "0.0.0"
+  ) //\todo remove this ? what does this affect ? is anyone using it ?
```
Comment (whiteshadow-gr) [inline src/main/scala/org/hatdex/dex/api/services/DexOffers.scala:31]: This is a dinosaur known with many names. `Datatrader` was the latest one. Before that was known as [Databuyer](https://github.com/dataswift/DataBuyer)

Not sure on the state of it or any future plans. I am not sure if it's live somewhere even. We had some discussions with a university to spin something off based on Databuyer but that was years ago. So @augustinas should chip in here.

File: src/main/scala/org/hatdex/dex/apiV2/services/DexStats.scala [hunk 2]
```diff
@@ -9,18 +9,17 @@
...
+import scala.concurrent.{ ExecutionContext, Future }
+
+import io.dataswift.models.dex.NamespaceStructure
+import io.dataswift.models.dex.json.DexJsonFormats
 import io.dataswift.models.hat.DataStats
+import org.hatdex.dex.apiV2.services.Errors.{ ApiException, DataFormatException }
 import play.api.Logger
 import play.api.http.Status._
-import play.api.libs.json.Json
+import play.api.libs.json.{Format, Json}
```
Comment (whiteshadow-gr) [inline src/main/scala/org/hatdex/dex/apiV2/services/DexStats.scala:20]: Minor: Formatting issue

File: src/main/scala/org/hatdex/dex/apiV3/services/DexStats.scala [hunk 3]
```diff
@@ -9,18 +9,17 @@
...
+import scala.concurrent.{ ExecutionContext, Future }
+
+import io.dataswift.models.dex.NamespaceStructure
+import io.dataswift.models.dex.json.DexJsonFormats
 import io.dataswift.models.hat.DataStats
+import org.hatdex.dex.apiV3.services.Errors.{ ApiException, DataFormatException }
 import play.api.Logger
 import play.api.http.Status._
-import play.api.libs.json.Json
+import play.api.libs.json.{Format, Json}
```
Comment (whiteshadow-gr) [inline src/main/scala/org/hatdex/dex/apiV3/services/DexStats.scala:20]: minor: formatting issue

**Review by whiteshadow-gr — APPROVED** [review]

File: src/main/scala/org/hatdex/dex/api/services/DexOffers.scala (hunk 1 above)
Comment (benshaw) [reply src/main/scala/org/hatdex/dex/api/services/DexOffers.scala:31]: i think you missed the point, this function calls ```/api/offer/$offerId/claims``` but that endpoint does not exist

**Review by whiteshadow-gr — APPROVED** [review]

**Pull request merged**